src/java.base/share/classes/sun/security/validator/EndEntityChecker.java

```diff
@@ -116,7 +116,7 @@ class EndEntityChecker {
 
     // TLS key exchange algorithms requiring keyEncipherment key usage
     private static final Collection<String> KU_SERVER_ENCRYPTION =
-        Arrays.asList("RSA");
```

Comment:
I understand that you haven't modified the `KU_SERVER_KEY_AGREEMENT` on line 122 because it has 4 elements. However, these 2 nearby lines using different styles look a little strange to me. Why restrict the number to 0, 1, and 2. If we have defined methods with up to 9 arguments, does that mean the new type is suitable for 9 elements?

Reply:
The reason is that for the cases of length 0, 1, 2 we don't create underlying array as we do for larger number of elements. In other words when we replace `Arrays.asList()` with `List.of()` we win only for 0, 1 and 2 elements.